plugins/inputs/prometheus/metrics_receiver_test.go

```diff
@@ -110,3 +114,20 @@ func Test_metricAppender_Commit(t *testing.T) {
 	}
 	assert.Equal(t, expected, *pmb[0])
 }
+
+func Test_loadConfigFromFile(t *testing.T) {
+	os.Setenv("POD_NAME", "collector-1")
+	configFile := filepath.Join("testdata", "target_allocator.yaml")
+	logger := kitlog.NewLogfmtLogger(os.Stdout)
+	var reloader = func(cfg *config.Config) error {
+		logger.Log("reloaded")
+		return nil
+	}
+	taManager := createTargetAllocatorManager(configFile, logger, nil, nil)
```

Comment:
Could you add error handling to this function so that it is easier to tell if any error occured during creation of the TA Manager

Reply:
In golang create function usually do not return errors, instead they will return nil or return with a disabled shell, which this function does the latter.